src/ops.rs

```diff
@@ -1530,3 +1536,90 @@ fn op_run_status(
   });
   Box::new(future)
 }
+
+struct GetMessageFuture {
+  pub state: Arc<IsolateState>,
+}
+
+impl Future for GetMessageFuture {
+  type Item = Option<Buf>;
+  type Error = ();
+
+  fn poll(&mut self) -> Result<Async<Self::Item>, Self::Error> {
+    assert!(self.state.worker_channels.is_some());
+    match self.state.worker_channels {
+      None => panic!("expected worker_channels"),
+      Some(ref wc) => {
+        let mut wc = wc.lock().unwrap();
+        wc.1.poll()
+      }
+    }
+  }
+}
+
+fn op_worker_get_message(
```

Comment:
Seeing this does make me feel a bit stronger in favor of a unified messaging substrate instead of having distinct ones.

Reply:
We do have a unified messaging substrate, `getMessage` and `postMessage` are built on top of the lower-level op dispatch system.

I'd rephrase your suggestion: "`postMessage` should be the means that we dispatch ops."

I'm not so convinced of that. I think we have very particular concerns in the IPC system that the WebWorkers API doesn't address like synchronous responses, zero-copy buffers.